### PR TITLE
feat: add burnIn Configuration option (currently a no-op)

### PR DESCRIPTION
**Pull request by AtofStryker**
- Closes N/A

### Additional details
adds the `experimentalBurnIn` configuration option to the test runner, which defaults to a  `default` key of `3` and a `flaky` key of `5`. `boolean` options can be provided to enable defaults or disable `burnIn` completely. `default` and `flaky` can be any integer greater than `0`.

### PR Tasks

- [ ] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Review by ryanpei — APPROVED** [review]

File: packages/config/src/options.ts [hunk 1]
```diff
@@ -152,6 +152,14 @@ const driverConfigOptions: Array<DriverConfigOption> = [
     validation: validate.isStringOrArrayOfStrings,
     overrideLevel: 'any',
     requireRestartOnChange: 'server',
+  }, {
+    // burnIn config is only considered in run-mode while recording to cypress-cloud
+    name: 'burnIn',
```
Comment (MuazOthman) [inline packages/config/src/options.ts]: Shouldn't this be `experimentalBunIn` for now?

Comment (AtofStryker) [reply packages/config/src/options.ts]: is the goal to initially release this under an experimental flag?

Comment (MuazOthman) [reply packages/config/src/options.ts]: I believe so, here's the relevant doc in ClickUp: https://app.clickup.com/18033298/docs/h6amj-44600/h6amj-58751
@ryanpei can set me straight

Comment (AtofStryker) [reply packages/config/src/options.ts]: Got it. `burnIn` was cut over to `experimentalBurnIn` in [6f563ca](https://github.com/cypress-io/cypress/pull/27377/commits/6f563cafcce0f37333d26b86fb4a2054c9fd4da2). should be ready for a re review

**Comment by ryanpei** [comment]
@AtofStryker to be clear, `experimentalBurnIn` will have default setting to `false`? When it's experimental, we don't want to enable it by default.

**Review by MuazOthman — COMMENTED** [review]
All looks good, but do we want to default to `false` during the experiment?

**Comment by AtofStryker** [comment]
@ryanpei @MuazOthman experiment should now be disabled by default in [96ccc31](https://github.com/cypress-io/cypress/pull/27377/commits/96ccc316cfe27b35b8b48903783da54549b1b7a0)

**Review by MuazOthman — APPROVED** [review]
Thanks for taking care of it!

**Pull request merged**